content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+
+**TODO: replace this video with Griffin's cool sphere example**
+<video controls loop><source  src="shader_material.mp4" type="video/mp4"/></video>
+
+This is thanks to the new [`AsBindGroup`] trait / derive, which does all of the hard work of converting the material to GPU-compatible datatypes, writing them to the GPU, and creating the final [`BindGroup`]. The [`AsBindGroup`] trait is powerful: it supports combining multiple fields into the same uniform binding, configuring texture binding types (2d, 3d, filtering, etc), and more. For details, check out the [`AsBindGroup`] docs.
+
+All built in materials, such as the PBR [`StandardMaterial`] have been ported to use this new system. We strive to make Bevy "internals" the same as user code, and this is no exception! [`Material`] also works seamlessly with our more advanced shader features, such as [shader pipeline specialization](https://github.com/bevyengine/bevy/blob/v0.8.0/examples/shader/shader_defs.rs).
+
+There is also an equivalent [`Material2d`] trait, which enables custom materials in 2D.
+
+[`Material`]: https://docs.rs/bevy/0.8.0/bevy/pbr/trait.Material.html
+[`AsBindGroup`]: https://docs.rs/bevy/0.8.0/bevy/render/render_resource/trait.AsBindGroup.html
+[`BindGroup`]: https://docs.rs/bevy/0.8.0/bevy/render/render_resource/struct.BindGroup.html
+[`StandardMaterial`]: https://docs.rs/bevy/0.8.0/bevy/pbr/struct.StandardMaterial.html
+[`Material2d`]: https://docs.rs/bevy/0.8.0/bevy/sprite/trait.Material2d.html
+
+## Camera Driven Rendering
+
+<div class="release-feature-authors">authors: @cart</div>
+
+In previous versions of Bevy, [`Cameras`][`Camera`] were selected and run as a part of one "global" [`RenderGraph`]. There could only be one "active" camera of a given type, and that camera could only render to one target. The only way to render from multiple perspectives at the same time was to manually extend the render graph with duplicate logic. This was _full_ of complicated low-level renderer boilerplate that wasn't approachable for the average Bevy user.
+
+In **Bevy 0.8**, each [`Camera`] now configures what it renders, how it renders it, and what it renders to. An enabled camera will start a new run of a [`RenderGraph`] to a specified [`RenderTarget`]. The render graph defines [modular render logic](/news/bevy-0-6/#render-graphs-and-sub-graphs) for the given camera and the render target defines the window or texture the graph will render to.
+
+This makes scenarios that were previously hundreds (or thousands) of lines of code as simple as setting a few fields on the [`Camera`] entity:
+
+### Render to Textures
+
+Rendering a [`Camera`] to a texture is now a single line of code:
+
+```rust
+camera.target = RenderTarget::Image(image_handle);
+```
+
+This enables too many scenarios to count: portals, rendering UI to a texture and rendering it in 3d space, in-game security cameras, real-time player portraits rendered in UI widgets ... the sky is the limit!
```

Comment:
```suggestion
This enables too many scenarios to count: portals, rendering UI to a texture and rendering it in 3d space, in-game security cameras, real-time player portraits rendered in UI widgets... The sky is the limit!
```

Reply:
This style is counter to how I was taught / how I use them. I think both approaches are valid and I personally like my style better, so I'll leave this as is.

Reply:
I agree; I think this change is wrong.